Automattic/wp-calypso
# Switch Email and Domain Management Email Paths

Some users wish to buy or manage G Suite without necessarily buying or managing their domain on WordPress.com. Currently G Suite is only accessible via (and nested under) the Domains management screens and paths, i.e. `/domains/manage/email/foo.com`. 

We also check that you have a domain when a user tries to visit any of the email management screens directly.

This patch moves email management to its own top-level `/email/foo.com` path and allows it to work without a domain. No other functionality should be affected.

#### Changes

* Email is now managed from `/email` routes. A mostly complete list of these changes are below:
  * `domains/manage/email` *->* `/email`
  * `domains/manage/email/:site` *->* `/email/:site`
  * `domains/manage/add-gsuite-users/:site` ->`email/add-gsuite-users/:site`
  * `domains/manage/:domainName/add-gsuite-users/:site` ->`email/:domainName/add-gsuite-users/:site`
  * `domains/manage/email/:domainName/edit/:site` *->* `/email/:domainName/manage/:site`

#### Testing Instructions

*Prerequisites*
* A site with a domain and GSuite
* A site with a domain and without GSuite ( can be the same as the above site )  
* A site without a domain

*Testing New Routes*
1. Navigate to `/email/:site` for your site with GSuite
1. Confirm you are taken to the email management screen
1. Click one of the "Add GSuite Users" Button
1. Observe the new URL. Confirm it matches `/email/:domain/add-gsuite-users/:site`
1. Observe the page and confirm it is the Add G Suite Users  page 
1. Click the "Back" Button
1. Observe the new URL. Confirm that it matches `/email/:domain/manage/:site`
1. Navigate to `/email/:domain/forwarding/:site` for the site with a domain and without g Suite
1. Observe the page and confirm it is the Email Forwarding  page 
1. Click the Back Button
1. Observe the new URL. Confirm that it matches `/email/:domain/manage/:site`
1. Confirm that you are shown the G Suite CTA
1. Click "Add G Suite"
1. Confirm you are taken to the add G Suite users page again
1. Fill in the info and click continue
1. Confirm G Suite was correctly added to your cart

*Testing Legacy Redirects*
1. Navigate to `/domains/manage/email`
1. Confirm you are taken to `/email` and given a list of sites ( if you have more than one)
1. Navigate to `/domains/manage/email/:site` for your site with G Suite
1. Confirm you are taken to `/email/:site`
1. Navigate to `/domains/manage/:domain/email/:site`
1. Confirm you are taken to `/email/:domain/manage/:site`
1. Navigate to `/domains/manage/:domain/email-forwarding/:site
1. Confirm you are taken to `/email/:domain/forwarding/:site`
1. Navigate to `/domains/manage/add-gsuite-users/:site`
1. Confirm you are taken to `/email/:site`
1. Navigate to `/domains/manage/:domain/add-gsuite-users/:site`
1. Confirm you are taken to `/email/:domain/add-gsuite-users/:site`
